Source/Particles/WarpXParticleContainer.cpp

```diff
@@ -623,24 +613,14 @@ WarpXParticleContainer::DepositCharge (WarpXParIter& pti, RealVector const& wp,
         // Lower corner of tile box physical domain
         // Note that this includes guard cells since it is after tilebox.ngrow
         // Take into account Galilean shift
-        const amrex::Real cur_time = warpx.gett_new(lev);
         const amrex::Real dt = warpx.getdt(lev);
-        const amrex::Real time_of_last_gal_shift = warpx.time_of_last_gal_shift;
-        const amrex::Real time_shift_rho_old = (cur_time - time_of_last_gal_shift);
-        const amrex::Real time_shift_rho_new = (cur_time + dt - time_of_last_gal_shift);
-        amrex::Array<amrex::Real,3> galilean_shift;
+        amrex::Real time_shift_delta;
         if (icomp==0){
-            galilean_shift = {
-                m_v_galilean[0]*time_shift_rho_old,
-                m_v_galilean[1]*time_shift_rho_old,
-                m_v_galilean[2]*time_shift_rho_old };
+            time_shift_delta = 0;
         } else{
-            galilean_shift = {
-                m_v_galilean[0]*time_shift_rho_new,
-                m_v_galilean[1]*time_shift_rho_new,
-                m_v_galilean[2]*time_shift_rho_new };
+            time_shift_delta = dt;
```

Comment:
Would it work if we simply do the following?
```cpp
const amrex::Real time_shift_delta = (icomp == 0) ? 0._rt : dt;
```

Reply:
Thanks @EZoni Yes, that will work. I'll make the change. I've accepted all of your other suggestions as well.